src/response.rs

```diff
@@ -88,46 +88,32 @@ impl Response {
     }
 
     /// Get an HTTP header.
-    pub fn header(&self, name: &HeaderName) -> Option<&Vec<HeaderValue>> {
+    pub fn header(&self, name: impl Into<HeaderName>) -> Option<&HeaderValues> {
         self.res.header(name)
     }
 
     /// Remove a header.
-    pub fn remove_header(&mut self, name: &HeaderName) -> Option<Vec<HeaderValue>> {
+    pub fn remove_header(&mut self, name: impl Into<HeaderName>) -> Option<HeaderValues> {
         self.res.remove_header(name)
     }
 
     /// Insert an HTTP header.
-    pub fn set_header(
-        mut self,
-        key: http_types::headers::HeaderName,
-        value: impl AsRef<str>,
-    ) -> Self {
-        let value = value.as_ref().to_owned();
-        self.res
-            .insert_header(key, &[value.parse().unwrap()][..])
-            .expect("invalid header");
+    pub fn set_header(mut self, key: impl Into<HeaderName>, value: impl ToHeaderValues) -> Self {
+        self.res.insert_header(key, value);
         self
```

Comment:
Oh actually, we should probably remove the `Self` return type from all of these methods. They should strictly forward to `http-types` here.